src/pybkgmodel/data.py

```diff
@@ -501,23 +543,235 @@ def load_events(cls, file_name, cuts):
+        bool
+            True if fits file according to file extension
+        """
+
+        ext = Path(file_name)
+
+        try:
+            with fits.open(ext) as file:
+                pass
+            compatible = True
+        except OSError:
+            compatible = False
+
+        return compatible
+
+    @classmethod
+    def get_obs_id(cls, file_name):
+        """Function reads the observation id from the fits file.
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        int
+            Observation ID number of the fits file
+
+        Raises
+        ------
+        RuntimeError
+            In case Observation ID is not found or Events HDU layer is missing. In this case the file is not complient to GADF.
+        """
+
+        obs_id = None
+
+        with fits.open(file_name, memmap=False) as input_file:
+            try:
+                obs_id = int(input_file["EVENTS"].header['OBS_ID'])
+            except Exception as error:
+                raise RuntimeError(f'Can not find observations ID in {file_name}') from error
+
+        return obs_id
+
+    @classmethod
+    def load_events(cls, file_name):
+        """Function to read the events from the DL3 file and to compute the missing variables needed by pybkgmodel
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        dict:
+            A dictionary with the event properties: arrival time, direction, and energy.
+        """
+
+        data_names_mapping = {
+            'EVENT_ID': 'daq_event_number',
+            'RA': 'event_ra',
+            'DEC': 'event_dec',
+            'GAMMANESS': 'gammaness',
+            'ENERGY': 'event_energy',
+        }
+
+        with fits.open(file_name, memmap=False) as input_file, \
+            erfa_astrom.set(ErfaAstromInterpolator(1 * u.s)):
+            try:
+                evt_hdu = input_file["EVENTS"]
+                evt_head = evt_hdu.header
+                evt_data = Table.read(evt_hdu)
+
+                event_data = {}
+
+                for key, name in data_names_mapping.items():
+                    if key in evt_data.keys():
+                        event_data[name] = evt_data[key].quantity
+
+                        if not evt_data[key].unit:
+                            event_data[name] *= u.one
+
+                # Event times need to be converted from Instrument reference epoch
+                ref_epoch = astropy.time.Time(evt_head['MJDREFI'],
+                                              evt_head['MJDREFF'],
+                                              scale=evt_head['TIMESYS'].lower(),
+                                              format='mjd'
+                                              )
+
+                evt_time = evt_data['TIME'].quantity + ref_epoch
+                event_data['mjd'] = evt_time.mjd * u.d
```

Comment:
I am not sure, but I think this mjd will now depend on the timescale of the input file, and I guess you use this value internally and assume it's in UTC or some other fixed timescale?

Why are you storing the mjd in any case and not the full time object?